internal/services/storage/storage_data_lake_gen2_filesystem_resource.go

```diff
@@ -127,62 +131,84 @@ func resourceStorageDataLakeGen2FileSystem() *pluginsdk.Resource {
 }
 
 func resourceStorageDataLakeGen2FileSystemCreate(d *pluginsdk.ResourceData, meta interface{}) error {
-	accountsClient := meta.(*clients.Client).Storage.AccountsClient
-	client := meta.(*clients.Client).Storage.FileSystemsClient
-	pathClient := meta.(*clients.Client).Storage.ADLSGen2PathsClient
+	storageClient := meta.(*clients.Client).Storage
 	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	storageID, err := commonids.ParseStorageAccountID(d.Get("storage_account_id").(string))
+	filesystemName := d.Get("name").(string)
+
+	// Parse the storage_account_id which is a resource manager ID
+	accountResourceManagerId, err := commonids.ParseStorageAccountID(d.Get("storage_account_id").(string))
 	if err != nil {
 		return err
 	}
 
-	aceRaw := d.Get("ace").(*pluginsdk.Set).List()
-	acl, err := ExpandDataLakeGen2AceList(aceRaw)
+	// Confirm the storage account exists and retrieve its properties
+	account, err := storageClient.FindAccount(ctx, accountResourceManagerId.StorageAccountName)
 	if err != nil {
-		return fmt.Errorf("parsing ace list: %s", err)
+		return fmt.Errorf("retrieving Account %q for Data Lake Filesystem %q: %v", accountResourceManagerId.StorageAccountName, filesystemName, err)
```

Comment:
should these all be noted as Gen2?
```suggestion
		return fmt.Errorf("retrieving Account %q for Data Lake Gen2 Filesystem %q: %v", accountResourceManagerId.StorageAccountName, filesystemName, err)
```